packages/grid/src/GridRenderer.ts

```diff
@@ -2503,6 +2557,109 @@ export class GridRenderer {
+          if (index < leftVisibleIndex) {
+            return Math.round((index / columnCount) * barWidth);
+          }
+          if (index > rightVisibleIndex) {
+            const remainderRight =
+              (getOrThrow(visibleColumnXs, rightVisibleIndex) +
+                getOrThrow(visibleColumnWidths, rightVisibleIndex) -
+                barWidth) /
+              getOrThrow(visibleColumnWidths, rightVisibleIndex);
+
+            const rightTickSize =
+              (barWidth - (x + handleWidth)) /
+              (columnCount - rightVisibleIndex + remainderRight);
+
+            return Math.round(
+              barWidth - (columnCount - index + 1) * rightTickSize
+            );
+          }
+          return Math.round(
+            (getOrThrow(visibleColumnXs, index) / width) * handleWidth + x
+          );
+        };
+
+        const filteredRanges = [...selectedRanges].filter(
+          value => value.startColumn != null && value.endColumn != null
+        ) as NoneNullColumnRange[];
+
+        filteredRanges.sort(GridRenderer.compareColumns);
+
+        const mergedRanges: NoneNullColumnRange[] = [];
+
+        for (let i = 0; i < filteredRanges.length; i += 1) {
+          const range = filteredRanges[i];
+          const { startColumn, endColumn } = range;
+          if (i === 0) {
+            mergedRanges.push({ startColumn, endColumn });
+          } else if (
+            startColumn - 1 <=
+            mergedRanges[mergedRanges.length - 1].endColumn
+          ) {
+            mergedRanges[mergedRanges.length - 1].endColumn = Math.max(
+              mergedRanges[mergedRanges.length - 1].endColumn,
+              endColumn
+            );
+          } else {
+            mergedRanges.push({ startColumn, endColumn });
+          }
+        }
+
+        for (let i = 0; i < mergedRanges.length; i += 1) {
+          const range = mergedRanges[i];
+          if (
+            range.startColumn != null &&
+            range.endColumn != null &&
+            (range.startColumn !== cursorColumn ||
+              range.endColumn !== cursorColumn)
+          ) {
+            const tickX = Math.round(
+              (range.startColumn / columnCount) * barWidth
+            );
+            const tickWidth = Math.max(
+              1,
+              Math.round(
+                ((range.endColumn + 1) / columnCount) * barWidth - tickX
+              )
+            );
+            const trackHeight = hScrollBarSize - scrollBarCasingWidth;
+            context.fillRect(
+              tickX,
+              y + scrollBarCasingWidth + Math.round(trackHeight / 3),
+              tickWidth,
+              Math.round(trackHeight / 3)
+            );
+          }
+        }
+
+        // Current Active Tick
+        if (cursorColumn != null) {
+          // const tickX = Math.round((cursorColumn / columnCount) * barWidth);
+          const tickX = getTickX(cursorColumn);
+          const tickWidth = 2;
```

Comment:
You shouldn't be hardcoding the tick width, you should be drawing for the "width" of the column, e.g.:

```
const startTickX = getTickX(cursorColumn);
const endTickX = getTickX(cursorColumn+1);
```

Reply:
Don wanted active tick to be 2px always